chore: refactor ai to use zod schemas

jeremydw opened the pull request.

jeremydw commented on packages/root-cms/ui/components/ChatBar/ChatBar.tsx:
```diff
@@ -0,0 +1,326 @@
+  | TextMessageBlock;
+
+const USE_DEBUG_STRING = false;
+const DEBUG_STRING = `Lorem ipsum 🥕 dolor sit amet.
```
I moved this component from the inline definition in `AIPage`, git doesn't show it, but the original code is here: https://github.com/blinkk/rootjs/blob/main/packages/root-cms/ui/pages/AIPage/AIPage.tsx#L28 (Is this needed still? If not, I can delete it - I think it's a holdover from when you were first building out the typewriter effect)

stevenle commented:
yeah, i dunno what it is, the sparkle feels devoid of emotion for me and i think these AI-related features should have some emotional feeling attached to its experience. i can handle the icon change in a separate PR, i'll convert everything all at once.

jeremydw commented:
SG; I've:

1. Removed typewriter debug code
2. Changed the icon to the robot
3. Made a slight change to the usage of `root-cms.d.ts` in the prompt to harden it

stevenle commented:
i gave it some more thought, let's go back to the sparkles icon for now. i may want to have some guidelines about when to use sparkles vs. robot, e.g. sparkles when automating generated content, robot for "chat". i can clean that up in a separate PR.

jeremydw commented:
sounds good; i like the sparkles but agree it should feel more "actionable" (i.e. with a verb). maybe we can mix the sparkles with another icon (e.g. a pencil icon).

- setting the stroke width to 1.75 on the sparkles icon seems to make it look a little nicer
- i also polished some other styles as part of this update (screenshots below), and added the "reset" button to the "Edit JSON" modal to make it consistent with the "Edit with AI" modal.

i'll send a separate PR for fixing the rich text editor on save.

<img width="708" height="399" alt="image" src="https://github.com/user-attachments/assets/d62d57d1-703a-4670-b62f-5b59199b7e26" />

<img width="847" height="491" alt="image" src="https://github.com/user-attachments/assets/3188c83a-931f-4a9e-998d-2b164536eabe" />

<img width="1608" height="997" alt="image" src="https://github.com/user-attachments/assets/6539f75a-1181-42cb-934a-e99396ee801b" />
